### ISSUE TITLE
Collection of requirements for the plug-in "Discounts for Shopaholic"

### ISSUE THREAD

**Issue by kharanenka** [lovata/oc-shopaholic-plugin#50]

**Comment by remagalot** [comment]
Discount for:
Single item
Cart total
User discount cart item
User discount cart total
Discount for Brand
Discount for Category

**Comment by Kunilt** [comment]
I like how it's made in Prestashop, discounts are separeted to two groups: catalog price rules (these applies to prices visible in catalog) and cart rules (these applies to carts, ability to use like discount coupons if coupon code is provided when creating).

Catalog price rules:
- Applies to
* Products
* Categories
* Brands

- Conditions
* By user (Authorized users, guest users, by specific users, by custom user group (Buddies plugin also needs improvements too))
* Date (From, to) when discount is visible and applied

Cart rules:
- Applies to
* Product (Fixed / percent discount)
* Cart subtotal (Fixed / percent discount)
* Shipping price (Make free shipping)
* Free gift (Adds to cart free gift as a discount)

- Conditions
* By user (Authorized users, guest users, by specific users, by custom user group (Buddies plugin also needs improvements too))
* Date (From, to)
* Minimum order sum (with shipping / without shipping)
* If product is in cart (by specific products, by category, by brand)
* Discount quantity (let discount code use for first 100 successful orders)

**Comment by gbaumeister** [comment]
In addition to what Kunilt wrote. Extensibility: Registering your own conditions via code.

**Comment by pvullioud** [comment]
ans option to automatically apply the coupon to the cart.
<img src="https://monosnap.com/file/yCl8S9BasvjSqITl1m28QK0ITw5kNA.png">

**Comment by vosco88** [comment]
Hi just checking what are the plans for this plugin as it would be really useful for couple of my sites?

**Comment by kharanenka** [comment]
Hi @vosco88 ! We already have 3 promo plugins ready to release. We are now testing them and preparing the documentation. They will release in OctoberCMS marketplace soon

**Comment by vosco88** [comment]
great :) looking forward to that.

**Comment by kharanenka** [comment]
We released 3 plugins for promo discounts: [Discount for Shopaholic](https://octobercms.com/plugin/lovata-discountsshopaholic), [Campaigns for Shopaholic](https://octobercms.com/plugin/lovata-campaignsshopaholic), [Coupons for Shopaholic](https://octobercms.com/plugin/lovata-couponsshopaholic) plugins.
We added 22 promo mechanisms in [Orders for Shopaholic] plugin. These can be discounts without checking any conditions or discounts with checking conditions for count of offer/positions/products, etc. Promo mechanisms can be used to change the total price of the order using different logic: applying of coupons, sales, loyalty system, personal discounts, etc.
